Mathlib/RingTheory/Ideal/Basic.lean

```diff
@@ -578,6 +578,31 @@ theorem pow_mem_of_mem (ha : a ∈ I) (n : ℕ) (hn : 0 < n) : a ^ n ∈ I :=
     (fun m _hm => (pow_succ a m).symm ▸ I.mul_mem_right (a ^ m) ha) hn
 #align ideal.pow_mem_of_mem Ideal.pow_mem_of_mem
 
+theorem pow_mem_of_pow_mem {m n : ℕ} (ha : a ^ m ∈ I) (h : m ≤ n) : a ^ n ∈ I := by
+  rw [← Nat.add_sub_of_le h, pow_add]
+  exact I.mul_mem_right _ ha
+
+theorem add_pow_mem_of_pow_mem_of_le {m n k : ℕ}
+    (ha : a ^ m ∈ I) (hb : b ^ n ∈ I) (hk : m + n ≤ k + 1) :
+    (a + b) ^ k ∈ I := by
+  rw [add_pow]
+  apply I.sum_mem
+  intro c _
+  apply mul_mem_right
+  by_cases h : m ≤ c
+  · exact I.mul_mem_right _ (I.pow_mem_of_pow_mem ha h)
+  · refine I.mul_mem_left _ (I.pow_mem_of_pow_mem hb ?_)
+    simp only [not_le, Nat.lt_iff_add_one_le] at h
+    rw [Nat.le_sub_iff_add_le, ← add_le_add_iff_right 1]
+    exact le_trans (by rwa [add_comm _ n, add_assoc, add_le_add_iff_left]) hk
+    rw [← add_le_add_iff_right 1]
+    exact le_trans h (le_trans (Nat.le_add_right _ _) hk)
```

Comment:
I can't immediately find it in the style guide but we prefer not to spawn subgoals during a rewrite like this. (In particular, any code that contains and `exact` statement which is followed by more code at the same indentation level is almost always not our preferred style.)
```suggestion
    have hck : c ≤ k := by
      rw [← add_le_add_iff_right 1]
      exact le_trans h (le_trans (Nat.le_add_right _ _) hk)
    rw [Nat.le_sub_iff_add_le hck, ← add_le_add_iff_right 1]
    exact le_trans (by rwa [add_comm _ n, add_assoc, add_le_add_iff_left]) hk
```